DFC.App.ContactUs.Services.EmailService/NotifyEmailService.cs

```diff
@@ -0,0 +1,55 @@
+﻿using DFC.App.ContactUs.Data.Contracts;
+using DFC.App.ContactUs.Data.Models;
+using Microsoft.Extensions.Logging;
+using Notify.Exceptions;
+using SendGrid;
+using SendGrid.Helpers.Mail;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+
+namespace DFC.App.ContactUs.Services.EmailService
+{
+    [ExcludeFromCodeCoverage]
+    public class NotifyEmailService<TEmailRequestModel> : INotifyEmailService<TEmailRequestModel>
+      where TEmailRequestModel : class, IEmailRequestModel
+    {
+        private readonly ILogger<NotifyEmailService<TEmailRequestModel>> logger;
+        private readonly INotifyClientProxy notifyClientProxy;
+        private readonly NotifyOptions notifyOptions;
+
+        public NotifyEmailService(
+            ILogger<NotifyEmailService<TEmailRequestModel>> logger,
+            INotifyClientProxy notifyClientProxy,
+            NotifyOptions notifyOptions)
+        {
+            this.logger = logger;
```

Comment:
Shouldn't you check for null on these constructor interface params?

Reply:
Added